File: app/src/main/java/dev/ragnarok/fenrir/api/adapters/ProfileInfoResponceDtoAdapter.java
package dev.ragnarok.fenrir.api.adapters;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;

import java.lang.reflect.Type;

import dev.ragnarok.fenrir.api.model.VkApiProfileInfoResponce;

public class ProfileInfoResponceDtoAdapter extends AbsAdapter implements JsonDeserializer<VkApiProfileInfoResponce> {
    private static final String TAG = ProfileInfoResponceDtoAdapter.class.getSimpleName();

    @Override
    public VkApiProfileInfoResponce deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        if (!checkObject(json)) {
            throw new JsonParseException(TAG + " error parse object");
        }
        VkApiProfileInfoResponce info = new VkApiProfileInfoResponce();
        JsonObject root = json.getAsJsonObject();

        if (root.has("name_request")) {
            info.status = 2;
        } else {
            info.status = optInt(root, "changed", 0) == 1 ? 1 : 0;
        }
        return info;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/adapters/ProfileInfoResponceDtoAdapter.kt
package dev.ragnarok.fenrir.api.adapters

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import dev.ragnarok.fenrir.api.model.VKApiProfileInfoResponse
import java.lang.reflect.Type

class ProfileInfoResponceDtoAdapter : AbsAdapter(), JsonDeserializer<VKApiProfileInfoResponse> {
    @Throws(JsonParseException::class)
    override fun deserialize(
        json: JsonElement,
        typeOfT: Type,
        context: JsonDeserializationContext
    ): VKApiProfileInfoResponse {
        if (!checkObject(json)) {
            throw JsonParseException("$TAG error parse object")
        }
        val info = VKApiProfileInfoResponse()
        val root = json.asJsonObject
        if (root.has("name_request")) {
            info.status = 2
        } else {
            info.status = if (optInt(root, "changed", 0) == 1) 1 else 0
        }
        return info
    }

    companion object {
        private val TAG = ProfileInfoResponceDtoAdapter::class.java.simpleName
    }
}